drivers/adc/adc_mchp_xec.c

```diff
@@ -0,0 +1,334 @@
+/*
+ * Copyright (c) 2019 Intel Corporation.
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#define LOG_LEVEL CONFIG_ADC_LOG_LEVEL
+#include <logging/log.h>
+LOG_MODULE_REGISTER(adc_mchp_xec);
+
+#include <drivers/adc.h>
+#include <soc.h>
+#include <errno.h>
+#include <assert.h>
+
+#define ADC_CONTEXT_USES_KERNEL_TIMER
+#include "adc_context.h"
+
+#define XEC_ADC_VREF_ANALOG 3300
+#define XEC_ADC_DELAY_UNIT 40
+
+#define XEC_ADC_CHANNEL_COUNT 8
+#define XEC_ADC_CHANNELS_MASK 0x000000FF
+
+#define XEC_ADC_12_BITS 0b11
+#define XEC_ADC_10_BITS 0b10
+
+#define XEC_ADC_VREF_PAD 0b00
+#define XEC_ADC_VREF_GPIO 0b01
+#define XEC_ADC_VREF_MASK 0b11
+
+struct adc_xec_data {
+	struct adc_context ctx;
+	struct device *dev;
+	u16_t *buffer;
+	u16_t *repeat_buffer;
+	u32_t channels;
```

Comment:
This field also seems to be redundant. It is assigned a value copied from `sequence->channels` and never changed, so `ctx.sequence.channels` could be used instead.

Reply:
Removed.